### ISSUE TITLE
confirm training using backprop works

### ISSUE THREAD

**Issue by hamzasherik** [hamzasherik/Faouzia#12]
In this issue, we'll want to confirm training on a real dataset using the backprop implementation works (i.e. the model accurately predicts output from test set). Adjust implementation until solution that works is found. 

- [ ] confirmed model training on real dataset works and made any adjustments in code needed to make it work
- [ ] wrote unit tests for the backprop concrete method in Faouzia
- [ ] wrote unit tests for the train_model method in FaouziaTabularClassifier
- [ ] adjusted code to ensure unit test coverage and ensure test cases pass
